internal/venice-common/src/main/java/com/linkedin/venice/pubsub/manager/TopicManagerRepository.java

```diff
@@ -0,0 +1,72 @@
+package com.linkedin.venice.pubsub.manager;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * A repository of {@link TopicManager} instances, each associated with a specific PubSub region and cluster.
+ * This repository maintains one {@link TopicManager} for each unique PubSub bootstrap server address.
+ * While not mandatory, it is expected that each Venice component will have one and only one instance of this class.
+ */
+public class TopicManagerRepository implements Closeable {
+  private static final Logger LOGGER = LogManager.getLogger(TopicManagerRepository.class);
+  private final Map<String, TopicManager> topicManagers = new VeniceConcurrentHashMap<>();
+  private final TopicManagerContext topicManagerContext;
+  private final String localPubSubAddress;
+
+  public TopicManagerRepository(TopicManagerContext topicManagerContext, String localPubSubAddress) {
+    this.topicManagerContext = Objects.requireNonNull(topicManagerContext, "topicManagerContext cannot be null");
+    this.localPubSubAddress = Objects.requireNonNull(localPubSubAddress, "localPubSubAddress cannot be null");
+  }
+
+  // added in order to help with testing; visibility is package-private for testing purposes
+  TopicManager createTopicManager(String pubSubAddress) {
+    return new TopicManager(pubSubAddress, topicManagerContext);
+  }
+
+  /**
+   * By default, return TopicManager for local PubSub cluster.
+   */
+  public TopicManager getLocalTopicManager() {
+    return getTopicManager(localPubSubAddress);
+  }
+
+  public TopicManager getTopicManager(String pubSubAddress) {
+    return topicManagers.computeIfAbsent(pubSubAddress, k -> createTopicManager(pubSubAddress));
```

Comment:
The second param could be `this:: createTopicManager`, if you'd like.

Reply:
Yeah that's more concise. Updated.